### PR TITLE
mac-capture: Add SCStreamDelegate to stream to log capture errors

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Adds a SCStreamDelegate logging when SCK sources stop with an error.

### Motivation and Context

Having a delegate is needed for a future PR where we have to introduce some sort of UI feature that re-enables a stopped stream, as macOS 14 adds a UI for the user to stop SCK streams outside of OBS, or for when the computer goes to sleep (see https://github.com/obsproject/obs-studio/issues/8928).
I'm also low-key hoping (at least a tiny bit) that it could help us diagnose remaining freezing issues that still pop up very sporadically (assuming that they exit with error codes), and as such wouldn't mind this change making its way into a hotfix release (assuming other people think that's acceptable).

### How Has This Been Tested?

macOS 14
Put computer to sleep and (when waking it up again) saw the "Stream stopped with no capture source" message in the log.
Stopped the stream via the menu bar and saw the "User stopped stream" message.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/mac-capture/mac-screen-capture.m [hunk 1]
```diff
@@ -1334,6 +1334,30 @@ - (void)stream:(SCStream *)stream
...
+	NSString *errorMessage;
+	switch (error.code) {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 130000
+	case SCStreamErrorUserStopped:
+		errorMessage = @"User stopped stream.";
+		break;
+#endif
+	case SCStreamErrorNoCaptureSource:
+		errorMessage =
+			@"Stream stopped with no capture source. Did the computer go to sleep?";
```
Comment (PatTheMav) [inline plugins/mac-capture/mac-screen-capture.m:1348]: Error messages are not the place to voice hunches or assumption, they should be matter-of-fact only.

If context is required, additional context should be logged instead (i.e. the `NSError` object's error description itself or the stack).

Comment (gxalpha) [reply plugins/mac-capture/mac-screen-capture.m:1348]: I'll just remove it then, there is no more context to add from the NSError side as far as I can tell. The localized description just says "No displays or windows found to record" (original message on my machine: "Keine Displays oder Fenster für die Aufnahme gefunden"), and the stack is not very helpful helpful:
image

Comment (PatTheMav) [reply plugins/mac-capture/mac-screen-capture.m]: Bummer, but yeah that's kinda what the error code already spells out. Maybe rephrase to "Stream stopped as capture source was not found". "Capture source was lost" would also be possible, but that would make an assumption again.

Comment (gxalpha) [reply plugins/mac-capture/mac-screen-capture.m]: Changed to "Stream stopped as no capture source was not found".

**Review by PatTheMav — APPROVED** [review]
